core/src/filesystem/media/analyze_media_job.rs

```diff
@@ -0,0 +1,193 @@
+			},
+			AnalyzeMediaJobVariant::AnalyzeSeries(id) => {
+				Some(format!("Analyze series with id: {}", id))
+			},
+			AnalyzeMediaJobVariant::AnalyzeMediaGroup(ids) => {
+				Some(format!("Analyze media group with ids: {:?}", ids))
+			},
+		}
+	}
+
+	async fn init(
+		&mut self,
+		ctx: &WorkerCtx,
+	) -> Result<WorkingState<Self::Output, Self::Task>, JobError> {
+		let output = Self::Output::default();
+
+		// We match over the job variant to build a list of tasks to process
+		let tasks = match &self.variant {
+			// Single item is easy
+			AnalyzeMediaJobVariant::AnalyzeSingleItem(id) => {
+				vec![AnalyzeMediaTask::AnalyzeImage(id.clone())]
+			},
+			// For libraries we need a list of ids
+			AnalyzeMediaJobVariant::AnalyzeLibrary(id) => {
+				let library_media = ctx
+					.db
+					.media()
+					.find_many(vec![media::series::is(vec![series::library_id::equals(
+						Some(id.clone()),
+					)])])
+					.select(media::select!({ id }))
+					.exec()
+					.await
+					.map_err(|e| JobError::InitFailed(e.to_string()))?;
+
+				library_media
+					.into_iter()
+					.map(|media| AnalyzeMediaTask::AnalyzeImage(media.id))
+					.collect()
+			},
+			// We also need a list for series
+			AnalyzeMediaJobVariant::AnalyzeSeries(id) => {
+				let series_media = ctx
+					.db
+					.media()
+					.find_many(vec![media::series_id::equals(Some(id.clone()))])
+					.select(media::select!({ id }))
+					.exec()
+					.await
+					.map_err(|e| JobError::InitFailed(e.to_string()))?;
+
+				series_media
+					.into_iter()
+					.map(|media| AnalyzeMediaTask::AnalyzeImage(media.id))
+					.collect()
+			},
+			// Media groups already include a vector of ids
+			AnalyzeMediaJobVariant::AnalyzeMediaGroup(ids) => ids
+				.iter()
+				.map(|id| AnalyzeMediaTask::AnalyzeImage(id.clone()))
+				.collect(),
+		};
+
+		Ok(WorkingState {
+			output: Some(output),
+			tasks: tasks.into(),
+			completed_tasks: 0,
+			logs: vec![],
+		})
+	}
+
+	async fn execute_task(
+		&self,
+		ctx: &WorkerCtx,
+		task: Self::Task,
+	) -> Result<JobTaskOutput<Self>, JobError> {
+		let mut output = Self::Output::default();
+
+		match task {
+			AnalyzeMediaTask::AnalyzeImage(id) => {
+				// Get media by id from the database
+				let media_item = ctx
+					.db
+					.media()
+					.find_unique(media::id::equals(id.clone()))
+					.exec()
+					.await
+					.map_err(|e: prisma_client_rust::QueryError| {
+						JobError::TaskFailed(e.to_string())
+					})?
+					.ok_or_else(|| {
+						JobError::TaskFailed(format!(
+							"Unable to find media item with id: {}",
+							id
+						))
+					})?;
+
+				let path = media_item.path;
+				let page_count = get_page_count(&path, &ctx.config)?;
+				output.images_analyzed += 1;
+
+				// Update media item in database
+				let _ = ctx
+					.db
+					.media()
+					.update(media::id::equals(id), vec![media::pages::set(page_count)])
+					.exec()
+					.await?;
+				output.media_updated += 1;
```

Comment:
This update should probably only hit _if_ the page counts are actually different

Reply:
I went ahead and followed this rule in my latest commit - I check if there's metadata, and if there is and `page_count` doesn't match the one we calculated, update it. If there isn't, create new metadata with the new `page_count`.

I'd appreciate if you could check the logic there, it may not have been the most idiomatic prisma usage.